03_importing_and_exporting_data.Rmd

```diff
@@ -0,0 +1,332 @@
+output: 
+  distill::distill_article:
+    toc: true
+---
+
+
+
+:::obj
+
+**Learning objectives**
+ 
+ - Know the basics of how a function works
+ - Understand how to read (import) data into R
+ - Explore data within R using `View()`
+ - Understand how to write (export) data from R
+ - Understand how to read and write different data formats (e.g., excel, csv, shp, rds, dbf, access)
+ 
+:::
+
+<br>
+
+## What's a function?
+
+Functions are how work gets done in `R`, and before we jump into reading and writing data, we need to know how functions work because we will use functions to perform these tasks.
+
+A **function** takes any number of **arguments** and, performs some transformations, and returns an **output**. 
+
+For example, the `R` function `sum()` takes any number of numeric arguments and adds them together (recall you can view the documentation for `sum()` by entering `?sum`). Let's add 1 and 2 like so:  
+
+```{r function-example-sum}
+
+sum(1, 2)
+
+```
+
+In `R`, we can create sequences easily. If we wanted to create a sequence of numbers from 1 to 10, we can use the function `seq()`, which takes 3 arguments: `from` (start value), `to` (end value), and `by` (increment of the sequence). 
+
+```{r sequence}
+
+seq(0, 10, 1)
+
+```
+
+Convince yourself that creating sequences of arbitrary length is possible. Can you create a sequence from 0 to 1 million by an increment of 500?
+
+Because creating sequences incremented by 1 are so common, there's a special shorthand for these sequences, `1:10`. Let's take the sum of the sequence from 1 to 10 by providing it as an argument to the function `sum()`:
+
+```{r sequence-sum}
+
+sum(1:10)
+
+```
+
+
+`sum()` and `seq()` are two of many functions you'll encounter in `R`. Like all functions, they take inputs (arguments) and return an output. To take advantage of functions, we need to apply them to our data. Let's now use import and export functions in R to explore some water resources data.  
+
+<br>
+
+## Read (import) data
+
+Data come in many formats, and `R` has utilities for reading and writing all kinds of data. In this lesson, we'll explore some of the most common data formats you'll encounter in the wild, and the functions used to import these data.
+
+### csv (comma separated values) 
+
+The comma separated value, or csv, is a simple and effective way to store tabular data. To read a csv file, we first import the {`readr`} library, which contains the function `read_csv()`. We also load {`here`} in order to easily navigate our working directory. Let's read a file from our `data` folder that contains station information for groundwater level monitoring sites in California. 
+
+```{r read-csv}
+library(here)
+library(readr)
+
+# read the "stations" csv, save it as an object called "stations", and print the object
+stations <- read_csv(here("data", "gwl" ,"stations.csv"))
```

Comment:
might add a comment here mentioning "relative paths" and how we used `here()`, but typing "`data/`" + TAB will bring up anything under this *relative* path, hit enter, then TAB, `gwl` + enter, etc. It can be a powerful way to make sure your files are where they are supposed to be and named what you think they should be named...your computer will autocomplete correctly no matter what!

Reply:
I added a bit about relative and absolute paths. Feel free to add if there's something else missing. I actually didn't know you could use tab to view files -- that's super cool!